mock_hardware/src/generic_system.cpp

```diff
@@ -237,6 +243,83 @@ CallbackReturn GenericSystem::on_init(const hardware_interface::HardwareInfo & i
     initialize_storage_vectors(gpio_commands_, gpio_states_, gpio_interfaces_, info_.gpios);
   }
 
+  actuator_slowdown_ = std::stod(info_.hardware_parameters["actuator_slowdown"]);
+
+  const auto num_joints = std::accumulate(
+    info_.transmissions.begin(), info_.transmissions.end(), 0ul,
+    [](const auto & acc, const auto & trans_info) { return acc + trans_info.joints.size(); });
+
+  const auto num_actuators = std::accumulate(
+    info_.transmissions.begin(), info_.transmissions.end(), 0ul,
+    [](const auto & acc, const auto & trans_info) { return acc + trans_info.actuators.size(); });
+
+  // reserve the space needed for joint and actuator data structures
+  joint_interfaces_.reserve(num_joints);
+  actuator_interfaces_.reserve(num_actuators);
+
+  // TODO(Manuel) Add all types of transmissions
+  // create transmissions, joint and actuator handles
+  auto transmission_loader = transmission_interface::SimpleTransmissionLoader();
+
+  for (const auto & transmission_info : info_.transmissions)
+  {
+    // only simple transmissions are supported in this demo
+    if (transmission_info.type != "transmission_interface/SimpleTransmission")
+    {
+      std::string msg(
+        "Transmission " + transmission_info.name + " of type " + transmission_info.type.c_str() +
+        " not supported");
+      throw std::runtime_error(msg);
+    }
+
+    std::shared_ptr<transmission_interface::Transmission> transmission;
+
+    transmission = transmission_loader.load(transmission_info);
+
+    std::vector<transmission_interface::JointHandle> joint_handles;
+    for (const auto & joint_info : transmission_info.joints)
+    {
+      // this demo supports only one interface per joint
+      if (!(joint_info.state_interfaces.size() == 1 &&
+            joint_info.state_interfaces[0] == hardware_interface::HW_IF_POSITION &&
+            joint_info.command_interfaces.size() == 1 &&
+            joint_info.command_interfaces[0] == hardware_interface::HW_IF_POSITION))
+      {
+        std::string msg("Invalid transmission joint " + joint_info.name + " configuration");
+        throw std::runtime_error(msg);
+      }
+
+      const auto joint_interface =
+        joint_interfaces_.insert(joint_interfaces_.end(), InterfaceData(joint_info.name));
+
+      transmission_interface::JointHandle joint_handle(
+        joint_info.name, hardware_interface::HW_IF_POSITION,
+        &joint_interface->transmission_passthrough_);
```

Comment:
why do we need transmission pass_though?

Reply:
have to check this maybe using directly state_, command_ is fine but i think overriding was a problem

Reply:
need it because we would override the handle otherwise (Handle for actuator/joint needs a storage, but we have state and command. We could only give one of the two, so if we call the functions actuator_to_joint or joint_to_actuator we would involunatry override )